pyatlas/pyatlas/atlas.py

```diff
@@ -0,0 +1,1239 @@
+        )
+        self.atlas.relationMemberRoles.ParseFromString(zip_entry_data)
+
+    def _load_relationTags(self):
+        zip_entry_data = _read_zipentry(self.atlas_file, self._FIELD_RELATION_TAGS)
+        proto_relation_tags = autogen.ProtoPackedTagStore_pb2.ProtoPackedTagStore()
+        proto_relation_tags.ParseFromString(zip_entry_data)
+        self.atlas.relationTags = _get_packed_tag_store_from_proto(proto_relation_tags)
+
+    def _load_relationIndexToRelationIndices(self):
+        zip_entry_data = _read_zipentry(self.atlas_file,
+                                        self._FIELD_RELATION_INDEX_TO_RELATION_INDICES)
+        proto_multimap = autogen.ProtoLongToLongMultiMap_pb2.ProtoLongToLongMultiMap()
+        proto_multimap.ParseFromString(zip_entry_data)
+        self.atlas.relationIndexToRelationIndices = _get_dict_from_protolongtolongmultimap(
+            proto_multimap)
+
+    def _load_field(self, field_name):
+        if field_name not in self._FIELD_NAMES_TO_LOAD_METHODS:
+            raise KeyError('unrecognized field {}'.format(field_name))
+
+        # reflection code to get the appropriate load method for the field we are loading
+        method_name = self._FIELD_NAMES_TO_LOAD_METHODS[field_name]
+        load_method_to_call = getattr(self, method_name)
+        load_method_to_call()
+
+
+class _PackedTagStore(object):
+    """
+    Stores indexes into the global tag dictionary.
+    """
+
+    def __init__(self, dictionary):
+        self.keys = []
+        self.values = []
+        self.dictionary = dictionary
+
+    def set_dictionary(self, dictionary):
+        self.dictionary = dictionary
+
+    def __str__(self):
+        string = "KEYS:\n"
+        string += str(self.keys)
+        string += "\nVALUES:\n"
+        string += str(self.values)
+        return string
+
+    def to_key_value_dict(self, index):
+        if self.dictionary is None:
+            raise ValueError('dictionary is not set')
+
+        result = {}
+        if len(self.keys) == 0:
+            return result
+        key_array = self.keys[index]
+        value_array = self.values[index]
+
+        if len(key_array) != len(value_array):
+            raise IndexError('array length mismatch')
+
+        for key, value in zip(key_array, value_array):
+            result[self.dictionary.word(key)] = self.dictionary.word(value)
+
+        return result
+
+
+class _IntegerDictionary(object):
+    """
+    Integer string two-way dictionary.
+    """
+
+    def __init__(self):
+        self.word_to_index = {}
+        self.index_to_word = {}
+
+    def add(self, index, word):
+        if word in self.word_to_index:
+            return self.word_to_index[word]
+        self.word_to_index[word] = index
+        self.index_to_word[index] = word
+
+    def size(self):
+        return len(self.word_to_index)
+
+    def word(self, index):
+        # TODO this could throw a KeyError, how to handle?
```

Comment:
Something to address or ignore for now?

Reply:
I changed the way that class is used internally, so I believe that comment is outdated now. Good catch.